crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -411,22 +411,44 @@ impl<'input> CanOmitOptionalParenthesesVisitor<'input> {
 
     fn can_omit(self) -> bool {
         if self.max_priority_count > 1 {
-            false
-        } else if self.max_priority == OperatorPriority::Attribute {
-            true
-        } else if !self.any_parenthesized_expressions {
+            return false;
+        }
+
+        if self.max_priority == OperatorPriority::Attribute {
+            return true;
+        }
+
+        if !self.any_parenthesized_expressions {
             // Only use the more complex IR when there is any expression that we can possibly split by
-            false
-        } else {
-            // Only use the layout if the first or last expression has parentheses of some sort.
-            let first_parenthesized = self
-                .first
-                .is_some_and(|first| has_parentheses(first, self.source));
-            let last_parenthesized = self
-                .last
-                .is_some_and(|last| has_parentheses(last, self.source));
-            first_parenthesized || last_parenthesized
+            return false;
         }
+
+        if self.max_priority == OperatorPriority::Comparator {
+            // If at least one of the operands is a call expression...
+            if self.first.is_some_and(ruff_python_ast::Expr::is_call_expr)
+                || self.last.is_some_and(ruff_python_ast::Expr::is_call_expr)
+            {
+                // And neither of the expressions are (non-call) parenthesized expressions...
+                if self.first.is_some_and(|first| {
+                    first.is_call_expr() || !has_parentheses(first, self.source)
+                }) && self
+                    .last
+                    .is_some_and(|last| last.is_call_expr() || !has_parentheses(last, self.source))
+                {
+                    // Don't split.
+                    return false;
+                }
+            }
+        }
```

Comment:
I'm guessing this is not the right solution, but it passes the new fixtures. Should this instead be in `expr_compare.rs`?

Reply:
The method here mirrors https://github.com/psf/black/blob/2fd9d8b339e1e2e1b93956c6d68b2b358b3fc29d/src/black/lines.py#L884-L966 I had a quick glance and don't seem to find this logic

What I like to do is to add print statement to blacks code base and then run the formatting on a code snipped.

Reply:
@konstin any chance this already gets fixed by your Call chain PR?